app/models/grade_entry_form.rb

```diff
@@ -154,7 +154,7 @@ def from_csv(grades_data, overwrite)
         next
       elsif totals.empty?
         totals = row.drop(1)
-        if self.show_total && names.last == GradeEntryForm.human_attribute_name(:total)
+        if !self.show_total && names.last == GradeEntryForm.human_attribute_name(:total)
```

Comment:
@devan-srinivasan You're getting closer, but the logic in this function is correct. You should be fixing the test itself, I suspect it's setting up the data incorrectly.

Reply:
Aw man, okay. Will do.